technical-reference/reference-technical-articles/technical-doc-token-converter.md

```diff
@@ -0,0 +1,226 @@
+
+<figure><img src="../../.gitbook/assets/token_converters_sequence.svg" alt="Diagram showing the relationship among the harvesting, distribution, and converting of income"><figcaption>Relationship among the harvesting, distribution, and converting of income</figcaption></figure>
+
+Regarding the destinations of the Token converters:
+
+- `RiskFund` already includes the integration with the `Shortfall` contract. The USDT received in the conversions will be [auctioned off in the Shortfall contract](shortfall-and-auctions.md).
+- `XVSVaultTreasury` accumulates the XVS received in the conversions; those XVS will be sent to the `XVSVault` eventually via VIP, which will update also the APR in the `XVSVault`
+- `PrimeLiquidityProvider` accumulates [the Prime funds](prime.md), distributing them according to the speeds configured via VIP.
+
+# `AbstractTokenConverter`
+
+`AbstractTokenConverter` contract provides every feature to configure and convert the desired assets. The `RiskFundConverter` and the `SingleTokenConverters` extend this abstract contract (that could also be extended by any other contract in the future because it provides their features in an agnostic way).
+
+## Configuration of conversions
+
+There will be one configuration per pair `tokenAddressIn / tokenAddressOut`. Only authorized contracts (Governance) will be able to add or update these configurations. The configuration for a conversion includes:
+
+- `tokenAddressIn` and `tokenAddressOut`: address of the token accepted by the converter in the conversion, and the address of the token sent to the user/wallet from the converter in the conversion
+- `incentive`: percentage used to increase the amount finally sent to the user (see below).
+- `conversionAccess` (see more about private conversions below):
+    - `NONE`: Conversion is disabled for the pair
+    - `ALL`: Conversion is enabled for private conversion and users
+    - `ONLY_FOR_CONVERTERS`: Conversion is enabled only for private conversion
+    - `ONLY_FOR_USERS`: Conversion is enabled only for users
+
+For example, in the `XVSVaultConverter` there would be a configuration entry with these values:
+
+- `tokenAdressIn`: XVS token address (the converter accepts XVS)
+- `tokenAddressOut`: BTCB token address (the converter offers BTCB)
+- `incentive`: 0 (initially there won't be any incentive)
+- `conversionAccess`: ALL (everyone can perform conversions of XVS for BTCB in the `XVSVaultConverter`)
+
+## Incentives
+
+To incentivize conversions we allow an increase in the final amount sent out from the converter contract. Example:
+
+- Given:
+    - Conversion rate provided by the oracle: 1 XVS = 5 USDC
+    - Incentive: 10%
+- If the user sends 1 XVS to the contract, they will receive 5.5 USDC (5 USDC base + 0.5 USDC applying the incentive).
```

Comment:
```suggestion
To incentivize conversions we allow an increase in the final amount sent out from the converter contract. 

### Example

Given:
    - Conversion rate provided by the oracle: 1 XVS = 5 USDC
    - Incentive: 10%
    
If the user sends 1 XVS to the contract, they will receive 5.5 USDC (5 USDC base + 0.5 USDC applying the incentive).
```
We have a couple different formats for examples. I copied this one from the liquidation docs